python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -1275,6 +1275,16 @@ def _legacy_process_configs(self, config_data, entity_type, project_id, all_acti
             config_path, config_entity = config_data
             commands = []
 
+            # We don't need or want the descriptor object to be sent to the client.
+            # Since we're done with this config for this invokation, we can just
+            # delete it from the entity dict.
+            del config_entity["descriptor"]
```

Comment:
a possibly clearer and more robust way to do this is to build up a dictionary from scratch and include exactly the fields we want to pass to the user rather than subtracting them like this. That approach means we are only making assumptions about the `config_entity` keys that we want to operate on in this method. 

If for example the `descriptor` key was removed or a `foo_bar` key was added to `config_entity`, the logic here would break. 

IME, It is also generally safer to try to be more functional/immutable - avoid deleting things out of dictionaries in favour of building new data structures unless there is a serious performance hit - in that case we can comment why we are doing things that way.